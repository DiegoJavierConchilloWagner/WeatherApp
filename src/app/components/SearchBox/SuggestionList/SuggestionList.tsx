import { FC, MutableRefObject } from 'react';

import { IPlaceMapped } from '@common/interfaces';
import Suggestion from '../Suggestion/Suggestion';

import { StyledSuggestions } from '../styled';

interface Props {
	boxRef: MutableRefObject<null>;
	suggestions: IPlaceMapped[];
	handleClickOutside: () => void;
	handleSuggestionClick: (suggestion: IPlaceMapped) => void;
}

export const SuggestionList: FC<Props> = ({ boxRef, suggestions, handleClickOutside, handleSuggestionClick }) => {
	return (
		<StyledSuggestions ref={boxRef}>
			{suggestions.map((sugg) => (
				<Suggestion
					key={sugg.id}
					suggestion={sugg}
					handleClickOutside={handleClickOutside}
					handleSuggestionClick={handleSuggestionClick}
				/>
			))}
		</StyledSuggestions>
	);
};
