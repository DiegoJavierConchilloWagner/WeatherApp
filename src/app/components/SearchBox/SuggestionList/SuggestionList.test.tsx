import renderWithThemeAndProvider from '@common/utils/renderWhitThemeAndProvider';
import { SuggestionList } from './SuggestionList';

describe('Example SuggestionList', () => {
	const ref: React.MutableRefObject<null> = {
		current: null
	};
	it('should render correcty', () => {
		const wrapper = renderWithThemeAndProvider(
			<SuggestionList
				boxRef={ref}
				suggestions={[]}
				handleClickOutside={jest.fn()}
				handleSuggestionClick={jest.fn()}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
