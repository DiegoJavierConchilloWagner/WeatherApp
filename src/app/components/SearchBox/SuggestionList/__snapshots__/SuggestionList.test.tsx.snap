// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example SuggestionList should render correcty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  background: #0d0f40;
  width: 50vw;
  top: 5.6rem;
  border-bottom-right-radius: 15px;
  border-bottom-left-radius: 15px;
  overflow: hidden;
  z-index: 2;
  opacity: 0.85;
}

@media (max-width:768px) {
  .c0 {
    width: 82vw;
    top: 7.4rem;
  }
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  background: #0d0f40;
  width: 50vw;
  top: 5.6rem;
  border-bottom-right-radius: 15px;
  border-bottom-left-radius: 15px;
  overflow: hidden;
  z-index: 2;
  opacity: 0.85;
}

@media (max-width:768px) {
  .c0 {
    width: 82vw;
    top: 7.4rem;
  }
}

<div>
    <div
      class="c0"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
